src/base/utils/os.cpp

```diff
@@ -0,0 +1,412 @@
+    if (!path.isEmpty())
+        return path;
+
+    path = pythonSearchReg(SYSTEM_32BIT);
+    if (!path.isEmpty())
+        return path;
+
+    path = pythonSearchReg(SYSTEM_64BIT);
+    if (!path.isEmpty())
+        return path;
+
+    // Fallback: Detect python from default locations
+    const QStringList dirs = QDir("C:/").entryList(QStringList("Python*"), QDir::Dirs, QDir::Name | QDir::Reversed);
+    foreach (const QString &dir, dirs) {
+        const QString path("C:/" + dir + "/");
+        if (QFile::exists(path + "python.exe"))
+            return path;
+    }
+
+    return QString();
+}
+
+bool Utils::OS::areAssociationsSet() noexcept
+{
+#ifndef WIN_HAS_DEFAULT_PROGRAMS_API
+    Q_ASSERT(QSysInfo::WindowsVersion < QSysInfo::WV_VISTA);
+    return (isTorrentFileAssocSet() && isMagnetLinkAssocSet());
+#else
+    Q_ASSERT(QSysInfo::WindowsVersion < QSysInfo::WV_WINDOWS8);
+
+    if (QSysInfo::WindowsVersion < QSysInfo::WV_VISTA)
+        return (isTorrentFileAssocSet() && isMagnetLinkAssocSet());
+
+    // Vista, Win7
+    BOOL isDefault = FALSE;
+    try {
+        const COMLib comLib {COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE};
+
+
+        auto pAssoc = comLib.createObject<IApplicationAssociationRegistration>(
+                          CLSID_ApplicationAssociationRegistration, nullptr, CLSCTX_INPROC);
+        if (pAssoc)
+            pAssoc->QueryAppIsDefaultAll(AL_EFFECTIVE, APP_REGISTRY_NAME, &isDefault);
+    }
+    catch (const std::runtime_error &) {}
+
+    return (isDefault == TRUE);
+#endif
+}
+
+void Utils::OS::setAssociations()
+{
+#ifndef WIN_HAS_DEFAULT_PROGRAMS_API
+    Q_ASSERT(QSysInfo::WindowsVersion < QSysInfo::WV_VISTA);
+
+    setTorrentFileAssoc(true);
+    setMagnetLinkAssoc(true);
+#else
+    Q_ASSERT(QSysInfo::WindowsVersion < QSysInfo::WV_WINDOWS8);
+
+    if (QSysInfo::WindowsVersion < QSysInfo::WV_VISTA) {
+        // Vista, Win7
+        const COMLib comLib {COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE};
+
+        auto pAssoc = comLib.createObject<IApplicationAssociationRegistration>(
+                           CLSID_ApplicationAssociationRegistration, nullptr, CLSCTX_INPROC);
+        if (!pAssoc || FAILED(pAssoc->SetAppAsDefaultAll(APP_REGISTRY_NAME)))
+            throw std::runtime_error("An error occurred while set application associations.");
+    }
+    else {
+        setTorrentFileAssoc(true);
+        setMagnetLinkAssoc(true);
+    }
+#endif
+}
+
+bool Utils::OS::isTorrentFileAssocSet()
+{
+    if (QSysInfo::WindowsVersion >= QSysInfo::WV_VISTA)
+        return false;
+
+    return (getAssoc(".torrent") == "qBittorrent.File.Torrent");
+}
+
+bool Utils::OS::isMagnetLinkAssocSet()
+{
+    if (QSysInfo::WindowsVersion >= QSysInfo::WV_VISTA)
+        return false;
+
+    return (getAssoc("magnet") == "qBittorrent.Url.Magnet");
+}
+
+void Utils::OS::setTorrentFileAssoc(bool value)
+{
+    Q_ASSERT(QSysInfo::WindowsVersion < QSysInfo::WV_VISTA);
```

Comment:
But it seems to be called from line 300 with the opposite condition.

Reply:
Now with fixed typo above it should be right.